pkg/sql/sem/tree/like_test.go

```diff
@@ -22,6 +22,45 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/types"
 )
 
+func TestLikeEscape(t *testing.T) {
+	testData := []struct {
+		expr      string
+		pattern   string
+		escape    string
+		matches   Datum
+		erroneous bool
+	}{
+		{``, `{`, string('\x7f'), DBoolFalse, false},
+		{``, `}`, `}`, DBoolFalse, false},
+		{``, `%%%%%`, ``, DBoolTrue, false},
+		{``, `%%%%%`, `\`, DBoolTrue, false},
+		{`a[b]`, `%[[_]`, `[`, DBoolTrue, false},
+		{`+`, `++`, `+`, DBoolTrue, false},
+		{`a`, `}`, `}`, DBoolFalse, true},
+		{`a{}%`, `%}}}%`, `}`, DBoolTrue, false},
```

Comment:
You need to add the test cases that the test script found here with newlines and other stuff to make sure we don't regress in the future, like when if reimplement this.